src/Storage/JsonFileFeed.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * JSON File Feed class.
+ *
+ * @package Automattic\WooCommerce\ProductFeedForOpenAI
+ */
+
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\ProductFeedForOpenAI\Storage;
+
+use Automattic\WooCommerce\ProductFeedForOpenAI\Feed\FeedInterface;
+
+// This file works directly with local files. That's fine.
+// phpcs:disable WordPress.WP.AlternativeFunctions
+
+/**
+ * In-memory feed storage.
+ *
+ * This class simply stores all data in memory and generates JSON at once.
+ */
+class JsonFileFeed implements FeedInterface {
+	/**
+	 * Indicates if there are previous entries in the feed.
+	 *
+	 * @var bool
+	 */
+	private $has_entries = false;
+
+	/**
+	 * The path to the feed file.
+	 *
+	 * @var string
+	 */
+	private $file_path;
+
+	/**
+	 * The file handle.
+	 *
+	 * @var resource|null
+	 */
+	private $file_handle = null;
+
+	/**
+	 * Start the feed.
+	 *
+	 * @return void
+	 */
+	public function start(): void {
+		$upload_dir = wp_upload_dir( null, true );
+		$directory  = $upload_dir['basedir'] . '/product-feeds/';
+
+		if ( ! is_dir( $directory ) ) {
+			mkdir( $directory, 0755, true ); // We need to reconsider this access.
+		}
+
+		// Rudimentary, can be changed in the future.
+		$i = 1;
+		while ( file_exists( $directory . 'openai-feed-' . $i . '.json' ) ) {
+			++$i;
+		}
+
+		$this->file_path   = $directory . 'openai-feed-' . $i . '.json';
```

Comment:
All of this is experimental for now, and work in progress. Please do not focus on it.